### ISSUE TITLE
Suggestion: replace core file according to CI version

### ISSUE THREAD

**Issue by tianhe1986** [kenjis/ci-phpunit-test#120]
I'm glad to use it for unittest,.It is very cool! Thank you!

My project uses CI 3.0.2 and It works very well with ci-phpunit-test till now, but I found there are some differences between Loader.php and Loader.php-3.0.1, so I'm not sure if there are some potential problems for their unmatched version. 

For some reason, I could not update my project with newest CI, and someone else may encounter the same situation.

So I suggest replacing core file according to CI version. It would be suitable to get the content of original 'CodeIgniter.php' ,parse the version, and then load matched core files.

A plus question, is there any way to contribute to ci-phpunit-test?  I'd like to help improving it.^_^

**Comment by kenjis** [comment]
CI 3.0.x means just minor bug fixes. So I believe there are no problems.
But I am also not sure. You're logically right.
I can say, you're the first person who points it out. But you still do not find any real problem.

I accept Pull Requests. See https://github.com/kenjis/ci-phpunit-test/blob/master/CONTRIBUTING.md

**Comment by tianhe1986** [comment]
Yes, you are right. It's just a possibility now. I'll provide more detailed feedback while encountering  real problems.

What would you do with CI's minor version upgrading,  which may add minor features or significant fixes ?

**Comment by kenjis** [comment]
When a new CI version is released, I have to provide new version of ci-phpunit-test if the core files below changes:
https://github.com/kenjis/ci-phpunit-test/tree/master/application/tests/_ci_phpunit_test/replacing/core
This is because ci-phpunit-test does not change your CI core files, but it has changed core files above and just loads them instead of the real CI core files.
Fortunately, core classes that are replaced are just 2 classes (Loader and Input).

But I think there are no significant changes in them. At least in 3.0.x.

After all, if CI changes, ci-phpunit-test have to change, too.

And if you write *unit* tests, the changes of CI does not effect your tests.

**Comment by apnmrv** [comment]
Hello everyone,

I've got the exception, while trying to apply ci-phpunit-test library to a just installed CI-project:
=================
$ ./phpunit --debug

An uncaught Exception was encountered

Type:        ArgumentCountError
Message:     Too few arguments to function CI_Hooks::__construct(), 0 passed

==================
I use:
1) $ ./phpunit --version
PHPUnit 7.2.4 by Sebastian Bergmann and contributors.

2) CI and  ci-phpunit-test --- the latest, downloaded from Github.

Please, help
